### DISCUSSION TITLE
建议页面标题动态显示为文章名称

### DISCUSSION THREAD

**Discussion by dianso** [gaowei-space/markdown-blog#36] [Ideas]
我看到github.com/soft98-top/markdown-blog这个fork的项目里增加了文章标题功能，打开哪个文章，标签栏就会显示文章名。

还有可以显示图片，目前md目录下的图片无法正常读取

期待下次更新啊

**Comment by gaowei-space** [comment]
get √

**Reply by dianso** [reply]
最近会更新吗？希望增加这个功能。

**Reply by gaowei-space** [reply]
这个月会抽时间搞下v1.1.2版本，大概有这些功能，最后一个就是你提的这个咯
![image](https://github.com/gaowei-space/markdown-blog/assets/10205742/197294a0-85a6-4629-9e06-cc5d8cadfa88)

**Reply by dianso** [reply]
> 这个月会抽时间搞下v1.1.2版本，大概有这些功能，最后一个就是你提的这个咯 ![image](https://user-images.githubusercontent.com/10205742/243275579-197294a0-85a6-4629-9e06-cc5d8cadfa88.png)

能把根目录设置为静态资源目录吗？有些文件需要放到根目录，比如sitemap.xml

**Reply by gaowei-space** [reply]
我看看

**Reply by dianso** [reply]
> 我看看

每天都来看，求更新啊~~~

另外是不是静态网站不支持搜索呢，我用了一个方法效率很低，就是用go后端将每篇文章中的文字索引，做成一个json文件，然后前端加载这个文件来实现搜索，效率很低。

**Reply by gaowei-space** [reply]
最近在忙其他的，没来的及，争取这周末搞下

**Reply by dianso** [reply]
求更新~😀

**Reply by dianso** [reply]
(☆▽☆)

**Comment by dianso** [comment]
现在这种字典排序ABC 123从上往下的

要是能在配置文件自定义排序就好了，就不需要1@人，A@地 这种了。

**Reply by gaowei-space** [reply]
自定义排序，你的意思是：把每篇文章名称，在配置文件中排好序，程序依据这个顺序来展示导航？

**Reply by dianso** [reply]
是啊

左边导航要是能用一个配置项目来自定义就好了，这个选项开启后，按照自定义顺序来排列.md文件和文件夹顺序。

或者是读取每个目录中的Sort.txt来实现自定义排序。md根文件夹下有Sort.txt文件，里面是文件名或者文件夹，一行一个，从上往下排序。

**Reply by gaowei-space** [reply]
嗯，了解了，不过这样配置好像也有点麻烦，需要维护一个配置文件，文章改了名还得改这个

**Reply by dianso** [reply]
是有点麻烦，字典排序默认打开。

再增加这个功能就好了，url里面的字母@这种不太美观

我现在用的屏幕是29寸，看起来页面文章内容好小，两边空了太多了。请问如何修改把文章区域弄大一点啊。

**Reply by gaowei-space** [reply]
调这个值就可以:
![image](https://github.com/gaowei-space/markdown-blog/assets/10205742/f0fb1371-b2d0-4f38-abef-0851ff0f6f5d)

**Reply by dianso** [reply]
求更新啊，大哥

期待下一次更新！

**Comment by dianso** [comment]
你博客证书过期了，应该是腾讯云申请的吧

**Comment by dianso** [comment]
问了问chatgpt

解决了网页标题栏显示md文件名

app.go

	bytes, err := os.ReadFile(mdfile)

下面

	tmp := strings.Split(f, "/")
	title := tmp[len(tmp)-1]
	ctx.ViewData("Title", title+" - "+Title)
	ctx.ViewData("Article", mdToHtml(bytes))
	ctx.View("index.html")

由于我的md文件名和文件夹都不包含@，就没处理这个

**Reply by gaowei-space** [reply]
get

**Reply by dianso** [reply]
作者你好，什么时候有空更新一下呀

**Reply by gaowei-space** [reply]
最近一直在忙一件人生大事，要等过了这段时间咯。

猜你对 golang 应该也挺感兴趣的，可以尝试开发开发噢

**Reply by dianso** [reply]
好的，知道了，你先忙重要的事情

我是个初学者，还在摸索中。
